cmake/modules/FindBOOST.cmake

```diff
@@ -78,14 +78,54 @@ IF(Boost_FOUND)
   SET(BOOST_VERSION
     "${BOOST_VERSION_MAJOR}.${BOOST_VERSION_MINOR}.${BOOST_VERSION_SUBMINOR}"
     )
+
+  #
+  # Test that Boost.Iostreams is usuable.
+  #
+  RESET_CMAKE_REQUIRED()
+  PUSH_CMAKE_REQUIRED("${DEAL_II_CXX_VERSION_FLAG}")
+  PUSH_CMAKE_REQUIRED("-L${Boost_LIBRARY_DIRS}")
+  SET(CMAKE_REQUIRED_LIBRARIES "-lboost_iostreams")
+  CHECK_CXX_SOURCE_COMPILES(
+    "
+    #include <string>
+    #include <boost/iostreams/device/back_inserter.hpp>
+    #include <boost/iostreams/filter/gzip.hpp>
+    #include <boost/iostreams/filtering_stream.hpp>
+
+    int main()
+    {
+      std::string decompressed_buffer;
+      char test[1] = {'c'};
+
+      boost::iostreams::filtering_ostream decompressing_stream;
+      decompressing_stream.push(boost::iostreams::gzip_decompressor());
+      decompressing_stream.push(boost::iostreams::back_inserter(decompressed_buffer));
+      decompressing_stream.write (test, 1);
+    }
+    "
+    DEAL_II_BOOST_IOSTREAMS_USUABLE
+    )
+  MESSAGE(STATUS "${DEAL_II_BOOST_IOSTREAMS_USUABLE}")
+  IF(${DEAL_II_BOOST_IOSTREAMS_USUABLE})
+    MESSAGE(STATUS "Boost.Iostreams is usuable.")
+  ELSE()
+    MESSAGE(STATUS "Boost.Iostreams is not usuable. "
+                   "Probabaly it was not built with zlib support")
+    SET(Boost_FOUND FALSE)
+    SET(Boost_LIBRARIES "")
+    SET(Boost_INCLUDE_DIRS "")
+  ENDIF()
+  RESET_CMAKE_REQUIRED()
+
```

Comment:
It would be nicer to refactor this check to the file `configure_boost.cmake` into a macro:
```
MACRO(FEATURE_TRILINOS_FIND_EXTERNAL var)
  ...
ENDMACRO()
```
(Have a look at `configure_2_trilinos.cmake` (and others).

That way we make sure to properly unset cache variables if the external boost library is unusable.